### PR TITLE
feat: improve authentication/registration user flow

### PR DESCRIPTION
**Pull request by wreality**
Improve user login/register flow by:
- Using apolloclient state to store/fetch login status
- Router guards to protect routes that require login
- Redirect to dashboard after successful registration

## Review / Validation steps
1. `Lando rebuild -y`
2. Goto /
-   [x] "Register | login" should be in the top right
3. Click Register
4. Create an account
- [x] Should be redirected to `/dashboard`
- [x] "Username" with dropdown should be in topright nav
- [x] Refreshing browser should still show logged in in topright nav
- [x] Clicking Logout should redirect to `/` with "Register | Login" in navbar
5.  Login with username and password at `/login`
- [x] Redirect to /dashboard
6. Logout
7. Goto `/account/profile`
- [x] Should be redirected to `/login` with banner about needing to login
8. Login
- [x] Should be redirected to `/account/profile` rather than `/dashboard`

### CODE REVIEW

File: client/src/components/mixins/appAuth.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import gql from "graphql-tag";
+import { get } from "lodash";
```
Comment (wreality) [inline client/src/components/mixins/appAuth.js:2]:
```suggestion

```
Unused import

File: client/src/components/mixins/appAuth.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        };
+      } catch (error) {
+        if (error.graphQLErrors) {
+          var errors = error.graphQLErrors
+            .map(e => e.extensions?.code ?? false)
+            .filter(Boolean);
+        }
+        return {
+          success: false,
+          errors: errors ?? [FAILURE_OTHER]
```
Comment (wreality) [inline client/src/components/mixins/appAuth.js:47]:
```suggestion
          errors: errors ?? ["FAILURE_OTHER"]
```

File: client/src/layouts/MainLayout.vue [hunk 3]
```diff
@@ -1,111 +1,39 @@
...
-      });
-    }
-  },
-  preFetch({ store, redirect, currentRoute }) {
-    if (!store.getters["auth/isLoggedIn"]) {
-      console.log();
-      redirect(`/login?redirect=${encodeURIComponent(currentRoute.path)}`);
-    }
-  }
+  methods: {}
```
Comment (wreality) [inline client/src/layouts/MainLayout.vue:30]:
```suggestion

```

File: client/src/layouts/PublicLayout.vue [hunk 4]
```diff
@@ -1,78 +1,29 @@
...
-    logout() {
-      this.$store.dispatch("auth/logout").then(() => {
-        this.$router.push("/");
-      });
-    }
-  },
-  mounted() {
-    this.$store.dispatch("auth/fetch");
-  }
+  methods: {}
```
Comment (wreality) [inline client/src/layouts/PublicLayout.vue:26]:
```suggestion

```

**Review by gmeben — CHANGES_REQUESTED** [review]
While testing, I noticed that the email input validates the user input before the user focuses the following username input, which means it yells at me about my email not being valid before I've finished typing it. Not an ideal experience.

**Comment by wreality** [comment]
Hmm. I thought I made it not do that. I'll try to replicate and fix it. Agree that's not an ideal experience. 

What browser in case it's not happening on chrome?

File: client/src/layouts/MainLayout.vue [hunk 5]
```diff
@@ -1,111 +1,38 @@
...
-        </div>
-        <div class="text-subtitle">Submission & Review System</div>
-      </div>
-      <q-img src="header-back.jpg" class="header-image absolute-top" />
-    </q-header>
-
+    <app-header />
     <q-drawer v-model="leftDrawerOpen" show-if-above content-class="bg-grey-1">
-      <div class="drawer-header"></div>
+      <div class="drawer-header">Stuff To Do</div>
```
[the remark this answers is not in the document]
Comment (wreality) [reply client/src/layouts/MainLayout.vue:5]: Good point. I'll add some content. At least links to profile and dashboard pages.

**Comment by wreality** [comment]
@gmeben Fixed both of the issues you pointed out.  Let me know if there's anything else.

**Comment by gmeben** [comment]
@wreality The email input validation issue appears to be fixed now as of testing in Chrome 88.0, Safari 14.0.2, and Firefox 85.0 on macOS 11.1.

**Comment by gmeben** [comment]
@wreality Update: the email input on the register page is fixed but it looks like the email input on the login page has the same problem.

**Review by gmeben — CHANGES_REQUESTED** [review]
Confirmed Working

* Acceptance test registration and authentication
* Drawer flyout placeholder

Needs Fixing

* Email input at `/login` still validates before change

Minor Issues

* There's a lone visible backslash (`/`) after the name inputs at `/account/profile`
* First Name and Last Name inputs at `/account/profile` should probably just be a single Name or Display Name input

**Comment by wreality** [comment]
Fixed the email field on the login page.

The /account/profile stuff will get addressed as part of #120

**Comment by wreality** [comment]
@gmeben just a reminder to mark this approved if you're good with it so I can merge.

**Comment by gmeben** [comment]
@wreality Sorry, I was busy reviewing #112. I'll spin this up and submit my re-review shortly

**Review by gmeben — APPROVED** [review]

**Pull request merged**